packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -234,7 +234,12 @@ function ChartsXAxis(inProps: ChartsXAxisProps) {
   // Skip axis rendering if no data is available
   // - The domain is an empty array for band/point scales.
   // - The domains contains Infinity for continuous scales.
-  if ((ordinalAxis && domain.length === 0) || (!ordinalAxis && domain.some(isInfinity))) {
+  // - The position is set to 'none'.
+  if (
+    (ordinalAxis && domain.length === 0) ||
+    (!ordinalAxis && domain.some(isInfinity)) ||
+    position === 'none'
```

Comment:
Why did we add this? I thought this logic was already in the PR that added `position: 'none'` as an option

Reply:
I'll extract it into another PR, but pretty much if you had `position:'none'` and used a `ChartsXAxis` (composition) it would still show up.

Reply:
https://github.com/mui/mui-x/pull/16727